packages/components/src/Gesture/withCalendarGesture.tsx

```diff
@@ -1,37 +1,37 @@
 /* eslint-disable react/no-multi-comp,class-methods-use-this */
-import React from 'react';
-import PropTypes from 'prop-types';
+import * as React from 'react';
 import keycode from 'keycode';
 import omit from 'lodash/omit';
+import { focusOn } from './focus';
 
 const FIRST = 0;
 const LAST = Number.POSITIVE_INFINITY;
 
-function focusOn(element) {
-	if (element) {
-		element.focus();
-	}
+export interface CalendarGestureProps {
+	onKeyDown: (e: React.KeyboardEvent<HTMLInputElement>, ref: any, item: any) => void;
+	goToPreviousMonth: (cb: () => void) => void;
+	goToNextMonth: (cb: () => void) => void;
 }
 
 /**
  * Select all items in current calendar
  */
-function getAllItems(calendarRef) {
-	return calendarRef.querySelectorAll('td > button[data-value]');
+function getAllItems(calendarRef: HTMLElement) {
+	return calendarRef.querySelectorAll<HTMLButtonElement>('td > button[data-value]');
 }
 
 /**
  * Focus on the item within the current calendar.
  * If the day index is out of the calendar's limits, it focuses on the limits.
  */
-function focusWithinCurrentCalendar(calendarRef, indexToFocus) {
+function focusWithinCurrentCalendar(calendarRef: HTMLElement, indexToFocus = 0) {
 	const allItems = getAllItems(calendarRef);
```

Comment:
Until the function `getAllItem` is typed, maybe you can type the var to avoid the several `as` after ?

Like
```suggestion
	const allItems: HTMLElement[] = getAllItems(calendarRef);
```